File: WQChartsDemo/Objc/ViewController/MainVC.m
// https://github.com/CoderWQYao/WQCharts-iOS
//
// MainVC.m
// WQCharts
//
// Created by WQ.Yao on 2020/01/02.
// Copyright (c) 2020年 WQ.Yao All rights reserved.
//

#import "MainVC.h"
#import "PolygonChartVC.h"
#import "PieChartVC.h"
#import "RadarChartVC.h"
#import "AxisChartVC.h"
#import "BarChartVC.h"
#import "LineChartVC.h"
#import "AreaChartVC.h"
#import "BizChartVC.h"

@interface MainVC () <UITableViewDataSource,UITableViewDelegate>

@property (nonatomic, strong) UITableView* tableView;
@property (nonatomic, strong) NSArray<NSDictionary*>* datas;

@end

@implementation MainVC

- (void)viewDidLoad {
    [super viewDidLoad];

    self.navigationItem.title = @"WQCharts";
    self.view.backgroundColor = Color_Block_BG;
    [self.view addSubview:self.tableView];
}

- (void)viewWillLayoutSubviews {
    [super viewWillLayoutSubviews];
    self.tableView.frame = self.view.bounds;
}

-(NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.datas.count;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 60;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    NSDictionary* data = self.datas[indexPath.row];
    NSString* cellID = NSStringFromClass(UITableViewCell.class);
    UITableViewCell* cell = [tableView dequeueReusableCellWithIdentifier:cellID] ?: [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:cellID];
    cell.backgroundColor = UIColor.clearColor;
    cell.contentView.backgroundColor = UIColor.clearColor;
    cell.textLabel.textColor = Color_White;
    cell.textLabel.text = data[@"title"];
    
    UIView* selectedBackgroundView = UIView.new;
    selectedBackgroundView.backgroundColor = Color_Block_Card;
    cell.selectedBackgroundView = selectedBackgroundView;
    return cell;
}

- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:NO];
    
    NSDictionary* data = self.datas[indexPath.row];
    Class vcClass = data[@"class"];
    UIViewController* vc = [[vcClass alloc] init];
    vc.title = data[@"title"];
    
    self.navigationItem.backBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:nil style:UIBarButtonItemStylePlain target:nil action:nil];
    [self.navigationController pushViewController:vc animated:YES];
}

- (UITableView *)tableView {
    if(!_tableView) {
        UITableView* tableView = [[UITableView alloc] init];
        tableView.backgroundColor = UIColor.clearColor;
        tableView.dataSource = self;
        tableView.delegate = self;
        _tableView = tableView;
    }
    return _tableView;
}

- (NSArray<NSDictionary *> *)datas {
    if(!_datas) {
        NSArray<NSDictionary*>* datas = @[
            @{@"title":@"PolygonChart",@"class":PolygonChartVC.class},
            @{@"title":@"PieChart",@"class":PieChartVC.class},
            @{@"title":@"RadarChart",@"class":RadarChartVC.class},
            @{@"title":@"AxisChart",@"class":AxisChartVC.class},
            @{@"title":@"BarChart",@"class":BarChartVC.class},
            @{@"title":@"LineChart",@"class":LineChartVC.class},
            @{@"title":@"AreaChart",@"class":AreaChartVC.class},
            @{@"title":@"BizChart",@"class":BizChartVC.class},
        ];
        _datas = datas;
    }
    return _datas;
}

@end

File: WQChartsDemo/Swift/ViewController/MainVC.swift
// 代码地址: https://github.com/CoderWQYao/WQCharts-iOS
//
// MainVC.swift
// WQCharts
//
// Created by WQ.Yao on 2020/01/02.
// Copyright (c) 2020年 WQ.Yao All rights reserved.
//

import UIKit

class MainVC: UIViewController, UITableViewDataSource, UITableViewDelegate {
  
    lazy open var tableView: UITableView = {
        let tableView = UITableView()
        tableView.backgroundColor = .clear
        tableView.dataSource = self
        tableView.delegate = self
        return tableView
    }()
    
    lazy open var datas: [NSDictionary] = {
        let datas: [NSDictionary] = [
            ["title":"PolygonChart","class":PolygonChartVC.self],
            ["title":"PieChart","class":PieChartVC.self],
            ["title":"RadarChart","class":RadarChartVC.self],
            ["title":"AxisChart","class":AxisChartVC.self],
            ["title":"BarChart","class":BarChartVC.self],
            ["title":"LineChart","class":LineChartVC.self],
            ["title":"FlowChart","class":FlowChartVC.self],
        ]
        return datas
    }()

    override func viewDidLoad() {
        super.viewDidLoad()
        
        navigationItem.title = "WQCharts"
        view.backgroundColor = Color_Block_BG
        view.addSubview(self.tableView)
    }
    
    override func viewWillLayoutSubviews() {
        super.viewWillLayoutSubviews()
        self.tableView.frame = self.view.bounds
    }
    
    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return datas.count
    }
      
    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 60
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let data = datas[indexPath.row]
        let cellID = NSStringFromClass(UITableViewCell.self)
        let cell: UITableViewCell = tableView.dequeueReusableCell(withIdentifier: cellID) ?? UITableViewCell.init(style: .default, reuseIdentifier: cellID)
        cell.backgroundColor = .clear
        cell.contentView.backgroundColor = .clear
        cell.textLabel?.textColor = Color_White
        cell.textLabel?.text = data["title"] as? String
        
        let selectedBackgroundView = UIView()
        selectedBackgroundView.backgroundColor = Color_Block_Card
        cell.selectedBackgroundView = selectedBackgroundView
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: false)
        
        let data = datas[indexPath.row]
        let vcClass = data["class"] as! UIViewController.Type
        let vc = vcClass.init()
        vc.title = (data["title"] as! String)
        
        self.navigationItem.backBarButtonItem = UIBarButtonItem.init(title: nil, style: .plain, target: nil, action: nil)
        self.navigationController?.pushViewController(vc, animated: true)
    }
    
}

